pandas/core/groupby/groupby.py

```diff
@@ -718,10 +718,18 @@ def _set_result_index_ordered(
         # the values/counts are repeated according to the group index
         # shortcut if we have an already ordered grouper
         if not self.grouper.is_monotonic:
-            index = Index(np.concatenate(self._get_indices(self.grouper.result_index)))
-            result.set_axis(index, axis=self.axis, inplace=True)
+            # row order is scrambled => sort the rows by position in original index
```

Comment:
let's change this here L720 to return on the comment case e.g.
```
if self.group_is_monotonic:
     result.set_axis(self.obj._get_axis(self.axis), axis=self.axis, inplace=True)
     return result
```
then you can outdent L721 on
you can consolidate the returns as well

Reply:
Done both